packages/rrweb-snapshot/src/snapshot.ts

```diff
@@ -820,8 +820,7 @@ function slimDOMExcluded(
         // (module)preload link
         (sn.tagName === 'link' &&
           (sn.attributes.rel === 'preload' ||
```

Comment:
```suggestion
          ((sn.attributes.rel === 'preload' && sn.attributes.as === 'script') ||
```
The rel="preload" and as="script" originally went together before https://github.com/rrweb-io/rrweb/pull/1128/files

Reply:
For `preload`, should we check: `as="script"` OR file extension == "js"? 

(Users don't always write HTML according to spec)

Reply:
I believe that the `as` part is necessary for the preload to 'work' correctly:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#as
So I'd be happy to conform to that.  
Otherwise we could add it to the following clause as in 

```
-        sn.attributes.rel === 'prefetch' &&
+        `['prefetch', 'preload'].includes(sn.attributes.rel) &&
```

but personally I'm not convinced that's necessary